### ISSUE TITLE
PNPM workspace: protocol is not removed before publish

### ISSUE THREAD

**Issue by StarpTech** [ghiscoding/lerna-lite#200]
### Describe the bug

As titled. This broke currently our release process :scream: 

```
lerna publish --conventional-commits --create-release github --allow-branch main --yes --message="chore(release): Publish [skip ci]"
```

### Reproduction

1. Create PNPM workspace
2. Use a dependency with workspace: protocol
3. Publish with lerna.
4. workspace: is still in the manifest.

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

### Environment Info

```shell
| Executable        | Version |
| ----------------- | ------- |
| `lerna --version` | v1.4.0 |
| `npm --version`   | VERSION |
| `yarn --version`  | VERSION |
| `node --version`  | VERSION |

---
OR simply run `npx lerna info` command
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/lerna-lite/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/ghiscoding/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by StarpTech** [comment]
Unfortunately, I had to switch back to lerna and use https://pnpm.io/workspaces#prefer-workspace-packages for published packages.

**Comment by ghiscoding** [comment]
It's not supposed to, it worked fine for me with a real pnpm project [ts-pnpm-monorepo](https://github.com/ghiscoding/ts-pnpm-monorepo) that I tested with and here's the [unpkg](https://unpkg.com/browse/@lerna-ws-test/bar@1.0.2/package.json). 

The PR for `workspace:` protocol is #148 the process is to call this method from `publish` command

https://github.com/ghiscoding/lerna-lite/blob/bffcb33677eb6bead47958110c6e2cc09ab8c28d/packages/publish/src/publish-command.ts#L569-L587

which then calls this line

https://github.com/ghiscoding/lerna-lite/blob/bffcb33677eb6bead47958110c6e2cc09ab8c28d/packages/core/src/package.ts#L282-L293

and I have a few unit tests for them as well, so it shouldn't publish at all any of the `workspace:` and I'm pretty sure there are other users having it in place also.  Did it fail or something? What was sent to unpkg? You didn't provide any sample of those failures

**Comment by StarpTech** [comment]
Yes, I'll try to provide more info. It could be also related to the pnpm-lock update after version bump because pnpm usually tries to use the workspace: protocol when updating dependencies.

**Comment by ghiscoding** [comment]
if that's the case then PR #199 might potentially be problematic then

**Comment by StarpTech** [comment]
I don't think so because you are updating the lock after version bump and removing it before publishing right?

**Comment by ghiscoding** [comment]
I'm not sure the word is really "removing" because technically speaking it's the package graph which is in memory that is simply being modified before publishing.

**Comment by ghiscoding** [comment]
I'm considering migrating the repo to a pnpm workspace to help with the support of `workspace:` protocol which is non-existent in npm. At the beginning I used npm so that anyone could contribute and also wanted to try npm workspaces but I'm not so impressed with it and now that I support workspace protocol it's even harder to support. If we do migrate to pnpm then we could use breakpoint directly in the code

**Comment by ghiscoding** [comment]
After some thought, I think I now why and it's just a theory but if I'm correct then it's because you're calling the shorter 1 liner command (version + publish) which fall in the last condition displayed below which is to create a new `VersionCommand()` instance

https://github.com/ghiscoding/lerna-lite/blob/bffcb33677eb6bead47958110c6e2cc09ab8c28d/packages/publish/src/publish-command.ts#L182-L190

however, and like I wrote in earlier post, I rely on the command name to know when to keep or remove the `workspace:` wording in this `updateLocalDependency()` function call 

https://github.com/ghiscoding/lerna-lite/blob/bffcb33677eb6bead47958110c6e2cc09ab8c28d/packages/core/src/package.ts#L250-L258

and the theory is that since it called `VersionCommand`, then it doesn't remove the `workspace:` wording because it think it is doing a `version` not a `publish`, I know it does the publish later but I'm not sure if it changes back the command name or not and if it doesn't then there's the bug... However like I said this is just a theory and to prove it, I'm adding a lot more helpful info in the next PR to include tarball temp location and `package.json` content so that we can debug this and provide helpful info to everyone via the `git-dry-run` flag. 

long story short, I'll look into fixing this but for a very likely quick fix you can use the 2 commands split, that is what I always use with Lerna (with extra config in [lerna.json](https://github.com/ghiscoding/lerna-lite/blob/main/lerna.json)) and that is how I tested the workspace protocol, so it looks like I did forget to test the 1 liner version/publish command.

https://github.com/ghiscoding/lerna-lite/blob/bffcb33677eb6bead47958110c6e2cc09ab8c28d/package.json#L19-L20

**Comment by StarpTech** [comment]
Thanks for the investigation. I'm going to try it out after the pnpm workspace lock fix.

**Comment by ghiscoding** [comment]
I gave this a try after merging the other 2 PRs and I can't seem to reproduce, at least not in dry-run mode. The last PR helps a lot by providing more helpful details about the tarball and what versions are being published, I tried with this simple demo

```ts
"dependencies": {
    "@lerna-lite/core": "workspace:^1.4.1",
    "@lerna-lite/info": "workspace:^1.4.1",
    "@lerna-lite/init": "workspace:^1.4.1",
    "@lerna-lite/publish": "workspace:*",
    "@lerna-lite/version": "workspace:^"
}
```
and ran the command you provided, in my use case I needed to turn off the new flags because npm isn't too happy with `workspace:` protocol but once I turn the lockfile-only thing, it seems to be working as expected

here's the command I ran locally
```ts
"dist-roll-publish-dry-run": "node ./packages/cli/dist/cli.js publish --conventional-commits --create-release github --allow-branch main --yes --no-sync-workspace-lock --no-manually-update-root-lockfile --git-dry-run",
```
and here's the tarball and publish details (again thanks to last PR you'll see a lot more useful info in dry-run mode)
```sh
lerna-lite notice === Tarball Details ===
lerna-lite notice name:          @lerna-lite/cli
lerna-lite notice version:       1.4.2
lerna-lite notice filename:      lerna-lite-cli-1.4.2.tgz
lerna-lite notice package size:  14.9 kB
lerna-lite notice unpacked size: 68.2 kB
lerna-lite notice shasum:        10fa5b50f555adc46f2c651e579cda8d04037370
lerna-lite notice integrity:     sha512-0hsvQ+NI0URkX[...]8hS6KLOWh5flA==
lerna-lite notice total files:   39
lerna-lite notice --- dry-run details ---
lerna-lite notice temp location: C:\Users\User\AppData\Local\Temp\16f72185-bd88-499d-89d2-1b5756b00df1\lerna-lite-cli-1.4.2.tgz
lerna-lite notice package name: @lerna-lite/cli
lerna-lite notice dependencies:
lerna-lite notice @lerna-lite/core    | ^1.4.2 
lerna-lite notice @lerna-lite/info    | ^1.4.2
lerna-lite notice @lerna-lite/init    | ^1.4.2
lerna-lite notice @lerna-lite/publish | 1.4.2
lerna-lite notice @lerna-lite/version | ^1.4.2
lerna-lite notice dedent              | ^0.7.0
lerna-lite notice dotenv              | ^16.0.1
lerna-lite notice import-local        | ^3.1.0
lerna-lite notice npmlog              | ^6.0.2
lerna-lite notice yargs               | ^17.5.1
lerna-lite notice
lerna-lite info dry-run> git checkout -- package.json packages\cli\package.json packages\core\package.json packages\exec\package.json packages\info\package.json packages\init\package.json packages\list\package.json packages\optional-cmd-common\package.json packages\publish\package.json packages\run\package.json packages\version\package.json
Successfully published:
 - @lerna-lite/cli@1.4.2
```
as you can see, `workspace:` is completely removed and so I can't seem to replicate the issue from my side. Some things to note are that I'm in `git-dry-run` mode and I merged the last 2 PRS, however I don't think any of these 2 points have any impact on the repro. 
Nonetheless, I would still suggest to use the 2 commands split that are `lerna version` and then later `lerna publish`, it's easier to follow and probably more predictable.

**Comment by StarpTech** [comment]
Hi @ghiscoding today I tried 1.5.0 and could reproduce it again. Here is the output of 

>  pnpm lerna publish --conventional-commits --git-dry-run --sync-workspace-lock

```
lerna-lite success published @wundergraph/nextjs 0.0.19
lerna-lite notice 
lerna-lite notice 📦  @wundergraph/nextjs@0.0.19
lerna-lite notice === Tarball Contents === 
lerna-lite notice 11.4kB LICENSE                      
lerna-lite notice 4.0kB  dist/handlebar.template.js   
lerna-lite notice 34.5kB dist/index.js                
lerna-lite notice 34.1kB dist/lib.js                  
lerna-lite notice 4.0kB  dist/template.controller.js  
lerna-lite notice 4.0kB  dist/template.js             
lerna-lite notice 1.3kB  package.json                 
lerna-lite notice 1.2kB  README.md                    
lerna-lite notice 4.1kB  dist/handlebar.template.d.ts 
lerna-lite notice 10.8kB dist/index.d.ts              
lerna-lite notice 10.8kB dist/lib.d.ts                
lerna-lite notice 264B   dist/template.controller.d.ts
lerna-lite notice 264B   dist/template.d.ts           
lerna-lite notice 70.7kB dist/tsconfig.tsbuildinfo    
lerna-lite notice === Tarball Details === 
lerna-lite notice name:          @wundergraph/nextjs                     
lerna-lite notice version:       0.0.19                                  
lerna-lite notice filename:      wundergraph-nextjs-0.0.19.tgz           
lerna-lite notice package size:  40.9 kB                                 
lerna-lite notice unpacked size: 191.5 kB                                
lerna-lite notice shasum:        121c98149e22684d3a02f06b8b0ac94ed75840ad
lerna-lite notice integrity:     sha512-KancdootLYKPd[...]kOlvT2hS2VGhQ==
lerna-lite notice total files:   14                                      
lerna-lite notice --- dry-run details ---
lerna-lite notice temp location: /tmp/212ba1fb-c274-4b7d-8b4a-669fde2cf36b/wundergraph-nextjs-0.0.19.tgz
lerna-lite notice package name: @wundergraph/nextjs
lerna-lite notice dependencies: 
lerna-lite notice @wundergraph/protobuf | ^0.91.1
lerna-lite notice handlebars            | ^4.7.7 
lerna-lite notice next                  | ^12.1.6
lerna-lite notice object-hash           | ^2.2.0 
lerna-lite notice react-ssr-prepass     | ^1.5.0 
lerna-lite notice devDependencies: 
lerna-lite notice @types/node        | ^17.0.32
lerna-lite notice @types/object-hash | ^1.3.4  
lerna-lite notice @types/react       | ^18.0.5 
lerna-lite notice @wundergraph/sdk   | ^0.92.2 
lerna-lite notice react              | ^18.1.0 
lerna-lite notice react-dom          | ^18.1.0 
lerna-lite notice typescript         | ^4.7.2  
lerna-lite notice 
lerna-lite success published @wundergraph/sdk 0.92.2
lerna-lite notice 
lerna-lite notice 📦  @wundergraph/sdk@0.92.2
lerna-lite notice === Tarball Contents === 
lerna-lite notice 11.4kB  LICENSE                                                               
lerna-lite notice 1.9kB   dist/codegen/templates/markdown/authentication.js                     
lerna-lite notice 1.8kB   dist/configure/authentication.js                                      
lerna-lite notice 3.8kB   dist/wunderctl/binarypath.js                                          
lerna-lite notice 4.9kB   dist/postman/builder.js                                               
lerna-lite notice 77B     dist/middleware/client.js                                             
lerna-lite notice 9.6kB   dist/graphql/configuration.js                                         
lerna-lite notice 10.8kB  dist/graphql/configuration.test.js                                    
lerna-lite notice 737B    dist/codegen/templates/markdown/configure_auth_providers.template.js  
lerna-lite notice 2.1kB   dist/wunderctl/exec.js                                                
lerna-lite notice 1.1kB   dist/graphql/federation.js                                            
lerna-lite notice 3.7kB   dist/codegen/templates/typescript/forms.js                            
lerna-lite notice 3.9kB   dist/codegen/templates/typescript/forms.tsx.template.js               
lerna-lite notice 3.5kB   dist/middleware/plugins/graphql.js                                    
lerna-lite notice 468B    dist/wunderctl/helper.js                                              
lerna-lite notice 2.3kB   dist/codegen/templates/typescript/hooks.js                            
lerna-lite notice 17.8kB  dist/middleware/plugins/hooks.js                                      
lerna-lite notice 6.5kB   dist/codegen/templates/typescript/hooks.template.js                   
lerna-lite notice 6.3kB   dist/codegen/index.js                                                 
lerna-lite notice 2.5kB   dist/codegen/templates/index.js                                       
lerna-lite notice 9.5kB   dist/codegen/templates/typescript/index.js                            
lerna-lite notice 42.3kB  dist/configure/index.js                                               
lerna-lite notice 632B    dist/cors/index.js                                                    
lerna-lite notice 30.0kB  dist/definition/index.js                                              
lerna-lite notice 1.1kB   dist/dotgraphqlconfig/index.js                                        
lerna-lite notice 591B    dist/env/index.js                                                     
lerna-lite notice 5.1kB   dist/index.js                                                         
lerna-lite notice 3.3kB   dist/linkbuilder/index.js                                             
lerna-lite notice 660B    dist/strings/index.js                                                 
lerna-lite notice 452B    dist/transformations/index.js                                         
lerna-lite notice 47.8kB  dist/v2openapi/index.js                                               
lerna-lite notice 270B    dist/version/index.js                                                 
lerna-lite notice 956B    dist/wunderctl/index.js                                               
lerna-lite notice 1.5kB   dist/wunderctlexec/index.js                                           
lerna-lite notice 18.2kB  dist/codegen/index.test.js                                            
lerna-lite notice 2.5kB   dist/codegen/templates/typescript/index.test.js                       
lerna-lite notice 9.6kB   dist/configure/index.test.js                                          
lerna-lite notice 687B    dist/definition/index.test.js                                         
lerna-lite notice 1.8kB   dist/linkbuilder/index.test.js                                        
lerna-lite notice 10.2kB  dist/transformations/index.test.js                                    
lerna-lite notice 3.6kB   dist/v2openapi/index.test.js                                          
lerna-lite notice 376B    dist/wunderctl/bin/install.js                                         
lerna-lite notice 649B    dist/wunderctl/install.js                                             
lerna-lite notice 3.3kB   dist/wunderctl/installer.js                                           
lerna-lite notice 1.8kB   dist/middleware/internal-client.js                                    
lerna-lite notice 2.2kB   dist/codegen/templates/typescript/internal.client.js                  
lerna-lite notice 898B    dist/codegen/templates/typescript/internal.client.template.js         
lerna-lite notice 15.1kB  dist/db/introspection.js                                              
lerna-lite notice 449B    dist/db/introspection.test.js                                         
lerna-lite notice 1.1kB   dist/codegen/templates/typescript/jsonschema.js                       
lerna-lite notice 528B    dist/codegen/templates/typescript/jsonschema.template.js              
lerna-lite notice 1.5kB   dist/codegen/templates/typescript/linkbuilder.js                      
lerna-lite notice 2.7kB   dist/codegen/templates/typescript/linkbuilder.template.js             
lerna-lite notice 397B    dist/wunderctl/logger.js                                              
lerna-lite notice 19.0kB  dist/definition/merge.js                                              
lerna-lite notice 42.0kB  dist/definition/merge.test.js                                         
lerna-lite notice 19.8kB  dist/definition/namespacing.js                                        
lerna-lite notice 15.6kB  dist/definition/namespacing.test.js                                   
lerna-lite notice 1.3kB   dist/codegen/templates/typescript/operations.js                       
lerna-lite notice 831B    dist/configure/operations.js                                          
lerna-lite notice 48.3kB  dist/graphql/operations.js                                            
lerna-lite notice 1.2kB   dist/codegen/templates/typescript/operations.template.js              
lerna-lite notice 41.6kB  dist/graphql/operations.test.js                                       
lerna-lite notice 10.4kB  dist/codegen/templates/typescript/react.hooks.template.js             
lerna-lite notice 5.4kB   dist/codegen/templates/typescript/react.js                            
lerna-lite notice 4.5kB   dist/codegen/templates/typescript/react.native.provider.template.js   
lerna-lite notice 4.1kB   dist/codegen/templates/typescript/react.provider.template.js          
lerna-lite notice 3.1kB   dist/graphql/renametypes.js                                           
lerna-lite notice 1.0kB   dist/graphql/renametypes.test.js                                      
lerna-lite notice 241B    dist/wunderctl/bin/run.js                                             
lerna-lite notice 2.1kB   dist/wunderctl/run.js                                                 
lerna-lite notice 7.2kB   dist/graphql/schema.js                                                
lerna-lite notice 1.7kB   dist/codegen/templates/typescript/server.js                           
lerna-lite notice 5.3kB   dist/middleware/server.js                                             
lerna-lite notice 1.3kB   dist/codegen/templates/typescript/server.template.js                  
lerna-lite notice 246B    dist/wunderctl/bin/uninstall.js                                       
lerna-lite notice 246B    dist/wunderctl/uninstall.js                                           
lerna-lite notice 2.0kB   dist/wunderctl/uninstaller.js                                         
lerna-lite notice 5.1kB   dist/codegen/templates/typescript/web.client.js                       
lerna-lite notice 16.4kB  dist/codegen/templates/typescript/web.client.template.js              
lerna-lite notice 234.4kB dist/generated/protobuf/wundernode_config.js                          
lerna-lite notice 2.3kB   package.json                                                          
lerna-lite notice 1.2kB   README.md                                                             
lerna-lite notice 285B    dist/codegen/templates/markdown/authentication.d.ts                   
lerna-lite notice 1.3kB   dist/configure/authentication.d.ts                                    
lerna-lite notice 212B    dist/wunderctl/binarypath.d.ts                                        
lerna-lite notice 558B    dist/postman/builder.d.ts                                             
lerna-lite notice 2.2kB   dist/middleware/client.d.ts                                           
lerna-lite notice 520B    dist/graphql/configuration.d.ts                                       
lerna-lite notice 11B     dist/graphql/configuration.test.d.ts                                  
lerna-lite notice 636B    dist/codegen/templates/markdown/configure_auth_providers.template.d.ts
lerna-lite notice 62B     dist/wunderctl/exec.d.ts                                              
lerna-lite notice 196B    dist/graphql/federation.d.ts                                          
lerna-lite notice 255B    dist/codegen/templates/typescript/forms.d.ts                          
lerna-lite notice 3.9kB   dist/codegen/templates/typescript/forms.tsx.template.d.ts             
lerna-lite notice 1.0kB   dist/middleware/plugins/graphql.d.ts                                  
lerna-lite notice 56B     dist/wunderctl/helper.d.ts                                            
lerna-lite notice 304B    dist/codegen/templates/typescript/hooks.d.ts                          
lerna-lite notice 606B    dist/middleware/plugins/hooks.d.ts                                    
lerna-lite notice 6.5kB   dist/codegen/templates/typescript/hooks.template.d.ts                 
lerna-lite notice 1.7kB   dist/codegen/index.d.ts                                               
lerna-lite notice 2.3kB   dist/codegen/templates/index.d.ts                                     
lerna-lite notice 1.3kB   dist/codegen/templates/typescript/index.d.ts                          
lerna-lite notice 7.1kB   dist/configure/index.d.ts                                             
lerna-lite notice 146B    dist/cors/index.d.ts                                                  
lerna-lite notice 8.4kB   dist/definition/index.d.ts                                            
lerna-lite notice 746B    dist/dotgraphqlconfig/index.d.ts                                      
lerna-lite notice 182B    dist/env/index.d.ts                                                   
lerna-lite notice 2.0kB   dist/index.d.ts                                                       
lerna-lite notice 1.6kB   dist/linkbuilder/index.d.ts                                           
lerna-lite notice 130B    dist/strings/index.d.ts                                               
lerna-lite notice 348B    dist/transformations/index.d.ts                                       
lerna-lite notice 193B    dist/v2openapi/index.d.ts                                             
lerna-lite notice 39B     dist/version/index.d.ts                                               
lerna-lite notice 161B    dist/wunderctl/index.d.ts                                             
lerna-lite notice 307B    dist/wunderctlexec/index.d.ts                                         
lerna-lite notice 241B    dist/codegen/index.test.d.ts                                          
lerna-lite notice 11B     dist/codegen/templates/typescript/index.test.d.ts                     
lerna-lite notice 11B     dist/configure/index.test.d.ts                                        
lerna-lite notice 11B     dist/definition/index.test.d.ts                                       
lerna-lite notice 11B     dist/linkbuilder/index.test.d.ts                                      
lerna-lite notice 11B     dist/transformations/index.test.d.ts                                  
lerna-lite notice 11B     dist/v2openapi/index.test.d.ts                                        
lerna-lite notice 64B     dist/wunderctl/bin/install.d.ts                                       
lerna-lite notice 84B     dist/wunderctl/install.d.ts                                           
lerna-lite notice 108B    dist/wunderctl/installer.d.ts                                         
lerna-lite notice 515B    dist/middleware/internal-client.d.ts                                  
lerna-lite notice 310B    dist/codegen/templates/typescript/internal.client.d.ts                
lerna-lite notice 811B    dist/codegen/templates/typescript/internal.client.template.d.ts       
lerna-lite notice 918B    dist/db/introspection.d.ts                                            
lerna-lite notice 11B     dist/db/introspection.test.d.ts                                       
lerna-lite notice 260B    dist/codegen/templates/typescript/jsonschema.d.ts                     
lerna-lite notice 438B    dist/codegen/templates/typescript/jsonschema.template.d.ts            
lerna-lite notice 271B    dist/codegen/templates/typescript/linkbuilder.d.ts                    
lerna-lite notice 2.7kB   dist/codegen/templates/typescript/linkbuilder.template.d.ts           
lerna-lite notice 72B     dist/wunderctl/logger.d.ts                                            
lerna-lite notice 4.7kB   dist/definition/merge.d.ts                                            
lerna-lite notice 11B     dist/definition/merge.test.d.ts                                       
lerna-lite notice 1.7kB   dist/definition/namespacing.d.ts                                      
lerna-lite notice 11B     dist/definition/namespacing.test.d.ts                                 
lerna-lite notice 260B    dist/codegen/templates/typescript/operations.d.ts                     
lerna-lite notice 2.9kB   dist/configure/operations.d.ts                                        
lerna-lite notice 3.0kB   dist/graphql/operations.d.ts                                          
lerna-lite notice 1.2kB   dist/codegen/templates/typescript/operations.template.d.ts            
lerna-lite notice 11B     dist/graphql/operations.test.d.ts                                     
lerna-lite notice 923B    dist/codegen/templates/typescript/react.d.ts                          
lerna-lite notice 10.6kB  dist/codegen/templates/typescript/react.hooks.template.d.ts           
lerna-lite notice 4.6kB   dist/codegen/templates/typescript/react.native.provider.template.d.ts 
lerna-lite notice 4.1kB   dist/codegen/templates/typescript/react.provider.template.d.ts        
lerna-lite notice 239B    dist/graphql/renametypes.d.ts                                         
lerna-lite notice 11B     dist/graphql/renametypes.test.d.ts                                    
lerna-lite notice 31B     dist/wunderctl/bin/run.d.ts                                           
lerna-lite notice 60B     dist/wunderctl/run.d.ts                                               
lerna-lite notice 170B    dist/graphql/schema.d.ts                                              
lerna-lite notice 299B    dist/codegen/templates/typescript/server.d.ts                         
lerna-lite notice 3.6kB   dist/middleware/server.d.ts                                           
lerna-lite notice 1.3kB   dist/codegen/templates/typescript/server.template.d.ts                
lerna-lite notice 55B     dist/wunderctl/bin/uninstall.d.ts                                     
lerna-lite notice 55B     dist/wunderctl/uninstall.d.ts                                         
lerna-lite notice 64B     dist/wunderctl/uninstaller.d.ts                                       
lerna-lite notice 1.0kB   dist/codegen/templates/typescript/web.client.d.ts                     
lerna-lite notice 17.0kB  dist/codegen/templates/typescript/web.client.template.d.ts            
lerna-lite notice 198.6kB dist/generated/protobuf/wundernode_config.d.ts                        
lerna-lite notice 97.7kB  dist/tsconfig.tsbuildinfo                                             
lerna-lite notice === Tarball Details === 
lerna-lite notice name:          @wundergraph/sdk                        
lerna-lite notice version:       0.92.2                                  
lerna-lite notice filename:      wundergraph-sdk-0.92.2.tgz              
lerna-lite notice package size:  180.7 kB                                
lerna-lite notice unpacked size: 1.2 MB                                  
lerna-lite notice shasum:        78e1f65286be0911eb86ed54f38e003144fae0fa
lerna-lite notice integrity:     sha512-AYgsjGGK78skF[...]huS2AB1oYzazg==
lerna-lite notice total files:   164                                     
lerna-lite notice --- dry-run details ---
lerna-lite notice temp location: /tmp/2a3b259b-1a78-4384-a894-b0929b3df80f/wundergraph-sdk-0.92.2.tgz
lerna-lite notice package name: @wundergraph/sdk
lerna-lite notice dependencies: 
lerna-lite notice @apollo/composition       | ^2.0.2           
lerna-lite notice @apollo/federation        | ^0.36.2          
lerna-lite notice @graphql-tools/schema     | ^8.3.10          
lerna-lite notice @prisma/generator-helper  | ^3.9.2           
lerna-lite notice @wundergraph/protobuf     | ^0.91.1          
lerna-lite notice @wundergraph/wunderctl    | workspace:^0.91.1
lerna-lite notice axios                     | ^0.26.1          
lerna-lite notice colors                    | ^1.4.0           
lerna-lite notice execa                     | 5.1.1            
lerna-lite notice fastify                   | ^3.27.1          
lerna-lite notice fastify-graceful-shutdown | ^3.1.0           
lerna-lite notice fastify-plugin            | ^3.0.1           
lerna-lite notice graphql                   | ^16.3.0          
lerna-lite notice graphql-helix             | ^1.12.0          
lerna-lite notice handlebars                | ^4.7.7           
lerna-lite notice js-yaml                   | ^4.1.0           
lerna-lite notice json-schema               | ^0.4.0           
lerna-lite notice lodash                    | ^4.17.21         
lerna-lite notice long                      | ^5.2.0           
lerna-lite notice object-hash               | ^2.2.0           
lerna-lite notice openapi-types             | ^10.0.0          
lerna-lite notice postman-collection        | ^4.1.1           
lerna-lite notice prettier                  | ^2.5.0           
lerna-lite notice protobufjs                | ^6.11.2          
lerna-lite notice swagger2openapi           | ^7.0.8           
lerna-lite notice devDependencies: 
lerna-lite notice @types/axios       | ^0.14.0  
lerna-lite notice @types/chai        | ^4.2.22  
lerna-lite notice @types/colors      | ^1.2.1   
lerna-lite notice @types/cors        | ^2.8.10  
lerna-lite notice @types/debug       | ^4.1.7   
lerna-lite notice @types/jest        | ^28.1.1  
lerna-lite notice @types/js-yaml     | ^4.0.0   
lerna-lite notice @types/json-schema | ^7.0.6   
lerna-lite notice @types/lodash      | ^4.14.178
lerna-lite notice @types/node        | ^14.18.16
lerna-lite notice @types/object-hash | ^1.3.4   
lerna-lite notice chai               | ^4.3.4   
lerna-lite notice jest               | ^28.1.0  
lerna-lite notice ts-jest            | ^28.0.4  
lerna-lite notice typescript         | ^4.7.2   
lerna-lite notice 
lerna-lite info dry-run> git checkout -- package.json packages/nextjs/package.json packages/sdk/package.json
```

As you can see the workspace protocol was removed properly except for SDK package (dependencies). This is because I don't manage `@wundergraph/wunderctl` with lerna. In my opinion, the prefix should be always removed.

**Comment by ghiscoding** [comment]
> This is because I don't manage `@wundergraph/wunderctl` with lerna. In my opinion, the prefix should be always removed nonetheless it is managed by Lerna or not.

That's most probably the problem, I can only assume that Lerna's code (not the code I added but the original Lerna's code) only ever loop into something that exists in its for loop but if you're not providing it then it probably won't do anything with it hence why it keeps the workspace. 

Removing the workspace: protocol for all might be doable but I can't work on it unless I have a repro. On my side I migrated Lerna-Lite to pnpm workspace yesterday and so I have a pnpm workspace to work with and the possibility to add breakpoints in the code. So if you want me to work on this, I need you to provide a repo or at the minimum the steps to do for testing this out, I really need a way to reproduce. Perhaps clone Lerna-Lite and modify it with your use case so that at least I get something to work with. Cheers

**Comment by StarpTech** [comment]
Hi @ghiscoding it's really simple to reproduce. Just remove one package from "packages" lerna.json but reference that dependency with the workspace: protocol in another package. If you publish the package, the workspace protocol isn't removed.

**Comment by StarpTech** [comment]
Everything else worked great! But this is a blocker for us.

**Comment by ghiscoding** [comment]
It's because of the following code shown below, the concept that Lerna uses is that it doesn't touch anything other than what it has to update, they have a filter with `localDependencies` and I believe the local ones are the ones showing up in your packages list and anything else falls under external dependencies which I guess are anything that aren't going to be updated

https://github.com/ghiscoding/lerna-lite/blob/a39c8d9ec3546638d08c5be33aa5e27231c4f297/packages/publish/src/publish-command.ts#L569-L587

I can make it work with some **exceptions**, it will only works correctly when providing a workspace: with a semver version, it won't work as intended on things like `workspace:*` since [npm-package-arg](https://www.npmjs.com/package/npm-package-arg) will interpret that as `latest`. So in your use case, I can get it to work but just make sure to always include a semver version when the package is outside of Lerna-Lite list of packages. 

So if your `package-1` external package is `workspace:1.2.3` then it would publish `"package-1": "1.2.3"`
But if your external package is `workspace:*` then it would publish `"package-1": "latest"` because [npm-package-arg](https://www.npmjs.com/package/npm-package-arg) parses it as "latest" and I don't want to have to add more code to fetch the real version number from registry. 

I think it's acceptable and I can add some non-blocking (not throwing) errors that can be seen in log output to provide some hints to possibly replace them

**Comment by StarpTech** [comment]
This is reasonable. LGTM.

**Comment by ghiscoding** [comment]
Published new release [v1.5.1](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.5.1) which includes the fix for this issue, hopefully that fixes it.

**Comment by StarpTech** [comment]
Great work and I could verify it in my current release. I found out that peerDeps aren't removed. Can we reopen this issue?
I made some comments on https://github.com/ghiscoding/lerna-lite/pull/219/files

**Comment by ghiscoding** [comment]
no need it's done

**Comment by StarpTech** [comment]
Thanks! Can you release it?

**Comment by ghiscoding** [comment]
sorry I can't, I'm traveling this week, that will have to be next weekend at the earliest

**Comment by StarpTech** [comment]
No problem, enjoy!
